torch/distributed/_tensor/ops/collectives_ops.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+# implement matrix related ops for distributed tensor
+
+import torch
+from torch.distributed._tensor.op_schema import OpSchema, OutputSharding
+from torch.distributed._tensor.ops.utils import register_prop_rule
+from torch.distributed._tensor.placement_types import DTensorSpec
+
+_c10d_functional = torch.ops._c10d_functional
+
+
+@register_prop_rule(_c10d_functional.all_to_all_single.default)
```

Comment:
Do we already have similar rules for other collective?

Reply:
Nope, there's no other rules